cmd/operator/main.go

```diff
@@ -256,3 +280,101 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=validatingwebhookconfigurations,verbs=get;list;watch;update;patch
+// +kubebuilder:rbac:groups="",namespace=keda,resources=secrets,verbs=get;list;watch;create;update;patch;delete
+
+// addCertificateRotation registers all needed services to generate the certificates and patches needed resources with the caBundle
+func addCertificateRotation(ctx context.Context, mgr manager.Manager, secretName, certDir, operatorServiceName, metricsServerServiceName, webhooksServiceName string) {
```

Comment:
Would be nice to move all certificate related stuff to a separate package/file, but this doesn't have to be done in this PR. But in the following one, dealing with the global certs solution.

Reply:
I plan to refactor the certs in operator and ms after this, I'll address this change there